### PR TITLE
apps x509: allow "-keyform engine" and passing PKCS#11 URL as -signkey

### PR DESCRIPTION
**Pull request by torbenh3**
openssl-1.1.0 has extended option checking, and rejects -keyform engine
and passing a PKCS#11 url to "-signkey" option. The actual code is ready
to take these.

Fixup the relevant Fields in x509_options[], and also fix the
optformat check in x509_main()
This is similar to bd6eba79d706 ("Fix small but important regression")
which does the same for CAkeyform.

Signed-off-by: Torben Hohn <torben.hohn@linutronix.de>

### CODE REVIEW

**Comment by torbenh3** [comment]
the patch is trivial, so i dont have to sign the CLA.

File: apps/x509.c [hunk 1]
```diff
@@ -72,7 +72,7 @@ const OPTIONS x509_options[] = {
     {"outform", OPT_OUTFORM, 'f',
      "Output format - default PEM (one of DER or PEM)"},
     {"out", OPT_OUT, '>', "Output file - default stdout"},
-    {"keyform", OPT_KEYFORM, 'F', "Private key format - default PEM"},
+    {"keyform", OPT_KEYFORM, 'f', "Private key format - default PEM"},
```
Comment (mattcaswell) [inline apps/x509.c]: 'E' looks more appropriate, i.e. PEM, DER or ENGINE (here and also for "cakeyform")

File: apps/x509.c [hunk 2]
```diff
@@ -217,7 +217,7 @@ int x509_main(int argc, char **argv)
                 goto opthelp;
             break;
         case OPT_KEYFORM:
-            if (!opt_format(opt_arg(), OPT_FMT_PEMDER, &keyformat))
+            if (!opt_format(opt_arg(), OPT_FMT_ANY, &keyformat))
```
Comment (mattcaswell) [inline apps/x509.c]: `OPT_FMT_PDE`? Here and for `cakeyform`.

Comment (torbenh3) [reply apps/x509.c]: do you want a separate patch for `cakeyform` ?

**Comment by levitte** [comment]
Uhmmmm, I don't quite understand why you needed to create a separate PR just for a correction.  Just commit the correction to the same branch and push.

**Comment by torbenh3** [comment]
sorry, this is the first time i have used githubs PR interface. I am used handling this kind of stuff
via mail.

File: apps/x509.c (hunk 2 above)
Comment (torbenh3) [reply apps/x509.c]: pushed V2 which is also rebased to master. But master fails in appveyor. so my V2 also fails

Comment (torbenh3) [reply apps/x509.c]: rebased to a fixed master. now it passes.

**Comment by bgermann** [comment]
These changes are now merged to master from other PRs. This PR can be closed.

**Comment by t8m** [comment]
> These changes are now merged to master from other PRs. This PR can be closed.

True. Closing.

**Pull request closed without merging**